GitExtensions.Extensibility/Settings/NumberSetting.cs

```diff
@@ -0,0 +1,182 @@
+﻿namespace GitExtensions.Extensibility.Settings;
+
+public class NumberSetting<T> : ISetting
+{
+    public NumberSetting(string name, T defaultValue)
+        : this(name, name, defaultValue)
+    {
+    }
+
+    public NumberSetting(string name, string caption, T defaultValue)
+    {
+        Name = name;
+        Caption = caption;
+        DefaultValue = defaultValue;
+    }
+
+    public string Name { get; }
+    public string Caption { get; }
+    public T DefaultValue { get; }
+    public Control? CustomControl { get; set; }
+
+    public ISettingControlBinding CreateControlBinding()
+    {
+        if (typeof(T) == typeof(int))
+        {
+            return new NumericUpDownBinding((this as NumberSetting<int>)!, CustomControl as NumericUpDown);
```

Comment:
Because it doesn't work...
![image](https://github.com/gitextensions/gitextensions/assets/4403806/2e74652f-d33a-4462-a00b-24946f941689)

Reply:
> Because it doesn't work...

... due to not to be expected technical restrictions of the compiler. `(T)variable` ought to be equivalent to `variable as T ?? throw new InvalidCastException(...)`.

Not in this case but basically, the `!` operator can be added such easily so that I fear it is added too carelessly.
If its accuracy is not obvious from a few preceding lines, it should be commented.
(And `null!` will always be an ugly hack to me.)

Reply:
> Not in this case but basically, the `!` operator can be added such easily so that I fear it is added too carelessly.
> If its accuracy is not obvious from a few preceding lines, it should be commented.


Trust me, I dislike `!` as much as you do for NRTA, and don't use it freely.
I'll add a comment.

> (And `null!` will always be an ugly hack to me.)

Aye, though it's often unavoidable when annotating legacy code.